freq/views.py

```diff
@@ -1,283 +1,913 @@
+    def enddate(self):
+        return self.to_date('end')
+
+    def get(self, request, uuid, *args, **kwargs):
+        self.set_session_value('startpage', 'measurement_point',
+                               "Meetpunt: " + self.request.GET['name'])
+        self.set_session_value('startpage', 'datepicker',
+                               {
+                                    'start': self.startdate,
+                                    'end': self.enddate
+                               })
+        self.set_session_value('startpage', 'start_js', request.GET['start'])
+        self.set_session_value('startpage', 'end_js', request.GET['end'])
+        self.set_session_value('startpage', 'uuid', uuid)
+        self.request.session['disabled']['trend_detection'] = 'enabled'
+        self.request.session.modified = True
+        self.set_session_value('trend_detection', 'active', True)
+        self.chart = ''
         return super().get(request, *args, **kwargs)
 
 
 class TrendDetectionView(BaseView):
+    freq_active = 'active'
     active = 'trend_detection'
     template_name = 'freq/trend_detection.html'
+    spinners = Spinner(
+        heading='Alpha (significance)',
+        title="alpha (significance of the t-test for changes in the " \
+                    "mean (0-1))",
+        min_=0.01,
+        step=0.01,
+        precision=2
+    )
+
+    dropdowns = DropDown(
+        heading= "Trend type",
+    )
+    spinner_max = 0.99
+    dropdown_options= [
+        "Both trends",
+        "Linear trend",
+        "Step trend"
+    ]
 
 
 class PeriodicFluctuationsView(BaseView):
-    active = 'periodic_fluctuation'
+    freq_active = 'active'
+    active = 'periodic_fluctuations'
     template_name = 'freq/periodic_fluctuations.html'
+    spinners = [
+        Spinner(
+            heading='Correlation lags',
+            title="Number of lags used for the correlogram computation",
+            number=0
+        ),
+        Spinner(
+            heading='Number of harmonics',
+            title="Number of harmonics to be removed from the series",
+            number=1
+        )
+    ]
+
+    @property
+    def spinner_max(self):
+        return int(self.length / 2)
+
+    @property
+    def spinner_1_value(self):
+        return self.request.session['periodic_fluctuations'].get(
+            'spinner_1', {'value': '0'})['value']
 
 
 class AutoRegressiveView(BaseView):
+    freq_active = 'active'
     active = 'autoregressive'
     template_name = 'freq/autoregressive.html'
+    spinners = Spinner(
+        heading='Number of periods',
+        title="Number of periods used in the training of the autoregressive " \
+               "model",
+        min_=1
+    )
 
+    @property
+    def spinner_max(self):
+        return int(0.3 * self.length)
 
-class SamplingFrequencyView(BaseView):
-    active = 'sampling'
-    template_name = 'freq/sampling_frequency.html'
 
+class BaseApiView(BaseViewMixin, APIView):
+    text_output = []
```

Comment:
Dit is een mutable object op de class. Als je in vier instanties `self.text_output.append("johooo")` doet is dat in elke instantie van de class zo! Je edit hetzelfde object :-)